cmake/kokkos_compiler_id.cmake

```diff
@@ -5,23 +5,19 @@ SET(KOKKOS_CXX_COMPILER_ID ${CMAKE_CXX_COMPILER_ID})
 SET(KOKKOS_CXX_COMPILER_VERSION ${CMAKE_CXX_COMPILER_VERSION})
 
 # Check if the compiler is nvcc (which really means nvcc_wrapper).
-EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
-                COMMAND grep nvcc
-                COMMAND wc -l
-                OUTPUT_VARIABLE INTERNAL_HAVE_COMPILER_NVCC
-                OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-
-STRING(REGEX REPLACE "^ +" ""
-       INTERNAL_HAVE_COMPILER_NVCC ${INTERNAL_HAVE_COMPILER_NVCC})
-
-
-IF(INTERNAL_HAVE_COMPILER_NVCC)
+GET_FILENAME_COMPONENT(KOKKOS_COMPILER_EXE ${CMAKE_CXX_COMPILER} NAME)
+STRING(REGEX MATCH "nvcc" HAVE_NVCC ${KOKKOS_COMPILER_EXE})
+IF (HAVE_NVCC)
+  STRING(REGEX MATCH "nvcc_wrapper" HAVE_NVCC_WRAPPER ${KOKKOS_COMPILER_EXE})
+  IF (NOT HAVE_NVCC_WRAPPER)
+    MESSAGE(FATAL_ERROR "Using NVCC compiler directly is not supported - must use nvcc_wrapper")
+  ENDIF()
   # SET the compiler id to nvcc.  We use the value used by CMake 3.8.
   SET(KOKKOS_CXX_COMPILER_ID NVIDIA CACHE STRING INTERNAL FORCE)
 
   # SET nvcc's compiler version.
-  EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version
+  EXECUTE_PROCESS(COMMAND ${CMAKE_CXX_COMPILER} --version --device
```

Comment:
why not just do this if we have HAVE_NVCC_WRAPPER and let people try use nvcc directly if they really want to?

Reply:
Yeah, we could do that. I hadn't considered the possibility of anyone using straight NVCC.